src/txpasslib/test/test_context.py

```diff
@@ -0,0 +1,56 @@
+from hypothesis import strategies as s
+from hypothesis import given
+from passlib.context import CryptContext
+from testtools import TestCase
+from testtools.matchers import Always, Equals, IsInstance
+from testtools.twistedsupport import succeeded
+from twisted._threads import createMemoryWorker
+
+from txpasslib.context import TxCryptContext
+from txpasslib.test.doubles import SynchronousReactorThreads
+from txpasslib.test.matchers import failed_with
+
+
+class ContextTests(TestCase):
+    """
+    Tests for `txpasslib.context.CryptContext`.
+    """
+    @given(password=s.text())
+    def test_hash_and_verify(self, password):
+        """
+        Hashing a password fires with the hash; verifying the password against
+        that hash succeeds, while verifying a different password fails.
+        """
+        reactor = SynchronousReactorThreads()
+        worker, perform = createMemoryWorker()
+        ctx = TxCryptContext(
+            CryptContext(schemes=['plaintext']),
+            reactor, worker)
+        d = ctx.hash(password)
+        perform()
+        self.assertThat(d, succeeded(Always()))
+        password_hash = d.result
```

Comment:
The main reason for the awkwardness here was the needed to call `perform()` after getting the deferred but before matching on things, so instead I made a matcher that does that! Let me know what you think of the new version. It's a bit `lispy)))))))))))))` as I often find happening with complex testtools matching, but I think the indentation makes it easy enough to follow.